benchmarks/StreamThroughput.cpp

```diff
@@ -72,85 +41,51 @@ class BM_Subscriber : public yarpl::flowable::Subscriber<Payload> {
     VLOG(2) << "BM_Subscriber destroy " << this;
   }
 
-  explicit BM_Subscriber(int initialRequest)
-      : initialRequest_(initialRequest),
-        thresholdForRequest_(initialRequest * 0.75),
-        received_(0) {
+  explicit BM_Subscriber(size_t requested) : requested_{requested} {
     VLOG(2) << "BM_Subscriber " << this << " created with => "
-            << "  Initial Request: " << initialRequest
-            << "  Threshold for re-request: " << thresholdForRequest_;
+            << "  Requested: " << requested_;
   }
 
-  void onSubscribe(yarpl::Reference<yarpl::flowable::Subscription>
-                       subscription) noexcept override {
+  void onSubscribe(
+      yarpl::Reference<yarpl::flowable::Subscription> subscription) override {
     VLOG(2) << "BM_Subscriber " << this << " onSubscribe";
+
     subscription_ = std::move(subscription);
-    requested_ = initialRequest_;
-    subscription_->request(initialRequest_);
+    subscription_->request(requested_);
   }
 
   void onNext(Payload element) noexcept override {
     VLOG(2) << "BM_Subscriber " << this
             << " onNext as string: " << element.moveDataToString();
 
-    received_.store(received_ + 1, std::memory_order_release);
-
-    if (--requested_ == thresholdForRequest_) {
-      int toRequest = (initialRequest_ - thresholdForRequest_);
-      VLOG(2) << "BM_Subscriber " << this << " requesting " << toRequest
-              << " more items";
-      requested_ += toRequest;
-      subscription_->request(toRequest);
-    };
-
-    if (cancel_) {
+    if (received_.fetch_add(1) == requested_ - 1) {
```

Comment:
True, though I'm tempted to be lazy and say that's not valid as `request(0)` isn't valid, hence we don't need to account for it :smile: